### ISSUE TITLE
vcrpy requirement in devtools is old and unavailable on some Linux distributions

### ISSUE THREAD

**Issue by major** [Azure/azure-sdk-for-python#20031]
- **Package Name**: azure-devtools
- **Package Version**: `HEAD`
- **Operating System**: Fedora 34
- **Python Version**: Python 3.9.6

**Describe the bug**
Fedora offers version 4.1.1 of vcrpy (released Oct 2020) but azure-devtools requires a much older 3.0.0 version that was released in 2019. Would it be possible to get this requirement updated?

**Comment by tjprescott** [comment]
Thanks for the report; we will look into it asap.

**Comment by tjprescott** [comment]
@lmazuel who would be the right person to evaluate and approve/reject this change?

**Comment by lmazuel** [comment]
Hi @major , there is several layers of answers to your question:
- We still support Python 2.7, which blocks us in 3.0.0. We tried conditional dependencies depending of Python2/3, but the recording format changed which would force us to record tests twice in two versions, in parallel folder or files names, while doubling the number of files on the repo. So we decided to stay in 3.0.0 until we drop Python 2.7, and we even forked it for bug fixes that we fought got no chances to be accepted by now ([see here](https://github.com/Azure/azure-sdk-for-python/tree/main/tools/vcrpy)).
- Parallel to that, we are migrated to a new test framework that will rely on a home made recorder which will be shared by all SDKs repos (C#, Java, C++, etc.), allowing us to totally get rid of vcrpy completely. Mostly relying on a Docker container mock server that you start and act as a recorder proxy. We are transitioning all our SDKs to this new system, and this should be done in the next few months to the end of the year.

Long story short, this means we have no plan to update, since vcrpy is about to disapear anyway. You can install our forked version for now from the repo if necessary, before you install devtools (installing devtools will be consider the installed version version compatible and would forward).
